testpar/t_cache_image.c

```diff
@@ -3937,50 +3657,28 @@ main(int argc, char **argv)
     }
 
     if (mpi_rank == 0) { /* create test files */
+        int i;
 
-        int   child_status;
-        pid_t child_pid;
-
-        child_pid = fork();
-
-        if (child_pid == 0) { /* this is the child process */
-
-            /* fun and games to shutup the compiler */
-            char  param0[32]   = "t_cache_image";
-            char  param1[32]   = "setup";
-            char *child_argv[] = {param0, param1, NULL};
-
-            /* we may need to play with the path here */
-            if (execv("t_cache_image", child_argv) == -1) {
-
-                HDfprintf(stdout, "execl() of setup process failed. errno = %d(%s)\n", errno,
-                          strerror(errno));
-                HDexit(1);
-            }
-        }
-        else if (child_pid != -1) {
-            /* this is the parent process -- wait until child is done */
-            if (-1 == waitpid(child_pid, &child_status, WUNTRACED)) {
-
-                HDfprintf(stdout, "can't wait on setup process.\n");
-            }
-            else if (!WIFEXITED(child_status)) {
+        HDfprintf(stdout, "Constructing test files: \n");
+        HDfflush(stdout);
 
-                HDfprintf(stdout, "setup process hasn't exitied.\n");
-            }
-            else if (WEXITSTATUS(child_status) != 0) {
+        i = 0;
+        while ((FILENAMES[i] != NULL) && (i < TEST_FILES_TO_CONSTRUCT)) {
+            HDfprintf(stdout, "   writing %s ... ", FILENAMES[i]);
+            HDfflush(stdout);
+            construct_test_file(i);
 
-                HDfprintf(stdout, "setup process reports failure.\n");
+            if (pass) {
+                HDprintf("done.\n");
+                HDfflush(stdout);
             }
             else {
-
-                HDfprintf(stdout, "testfile construction complete -- proceeding with tests.\n");
+                HDprintf("failed.\n");
+                HDexit(1);
```

Comment:
Better to use EXIT_SUCCESS and EXIT_FAILURE

Reply:
> Better to use EXIT_SUCCESS and EXIT_FAILURE

True, will update.